### PR TITLE
plat-544 Amazon Linux reverts timezone changes when running yum update

### PR DESCRIPTION
**Pull request by zhangjustin**
## Summary of changes

check `/etc/sysconfig/clock` and then modified `zone`

## How to Test

1. you need test on amazon linux
2. run ansible for amazon linux
3. check server date `date`

## Note
https://serverfault.com/questions/445984/prevent-amazon-ec2-time-zone-from-reverting-back-on-yum-update

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-time.html

https://forums.aws.amazon.com/thread.jspa?threadID=76293

/fyi:  @HouCoder ,  @mark-juwai : Please review.

### CODE REVIEW

**Review by mark-juwai — COMMENTED** [review]
```
[DEPRECATION WARNING]: Using tests as filters is deprecated. Instead of using `result|succeeded` instead use `result is succeeded`. This feature will be removed in version 2.9.
```

**Comment by mark-juwai** [comment]
Don't know how it exactly works.

**Review by mark-juwai — DISMISSED** [review]

**Comment by zhangjustin** [comment]
update,  now you can test with docker.
Reproduce error:
1. install docker and vagrant
2. run `vagrant up` on path `ansible-role-common/tests`
3. check `date`
4. run `sudo yum update`
5. check `date`

test fixed:
1. install docker and vagrant
2. run `vagrant up` on path `ansible-role-common/tests`
3. `vagrant ssh`  and create file `sudo touch  /etc/sysconfig/clock`
4. write `ZONE="Asia/Shanghai"`  to file `clock`
5. run `sudo yum update`
6. check `date`

**Comment by zhangjustin** [comment]
@tjoelsson , please review.

File: tasks/timezone.yml [hunk 1]
```diff
@@ -5,3 +5,15 @@
     dest: /etc/localtime
     state: link
     force: yes
+
+- name: clock file exist or not
+  shell: ls /etc/sysconfig/clock
```
Comment (tjoelsson) [inline tasks/timezone.yml]: Use [stat module](https://docs.ansible.com/ansible/latest/modules/stat_module.html) to check if file exists

File: tasks/timezone.yml [hunk 2]
```diff
@@ -5,3 +5,15 @@
...
+  ignore_errors: True
+  register: result
+
+- name: Set sysconfig clock
+  lineinfile:
+    dest: /etc/sysconfig/clock
```
Comment (tjoelsson) [inline tasks/timezone.yml]: ? single quotes '' around strings

File: tasks/timezone.yml [hunk 3]
```diff
@@ -5,3 +5,15 @@
     dest: /etc/localtime
     state: link
     force: yes
+
+- name: clock file exist or not
+  shell: ls /etc/sysconfig/clock
+  ignore_errors: True
+  register: result
```
Comment (tjoelsson) [inline tasks/timezone.yml]: Use a more meaningful name for the registered variable

**Comment by zhangjustin** [comment]
Updated, please review.

File: tasks/timezone.yml [hunk 4]
```diff
@@ -5,3 +5,15 @@
...
+  stat:
+    path: '/etc/sysconfig/clock'
+  register: sysconfig_clock
+
+- name: Set sysconfig clock
+  lineinfile:
+    dest: '/etc/sysconfig/clock'
+    regexp: '^ZONE='
+    line: 'ZONE="{{ timezone }}"'
+  when: sysconfig_clock.stat.exists is defined and sysconfig_clock.stat.exists == True
```
Comment (tjoelsson) [inline tasks/timezone.yml:19]: ? I think you can remove `== True`

**Review by tjoelsson — APPROVED** [review]

**Review by mark-juwai — APPROVED** [review]

**Comment by zhangjustin** [comment]
Merge, and don't need to deploy.

**Pull request merged**